src/Containers/book.js

```diff
@@ -1,46 +1,135 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
-
-function Book(props) {
-  const { book } = props;
-  const { title, category, progress, author } = book;
-
-  return (
-    <div className="notshown Book">
-      <div className="col">
-        <span className="fw-700">{category}</span>
-        <span>{title}</span>
-        <span>{author}</span>
-        <span className="sidemenu">
-          <span>Remove</span>
-          <span>Edit</span>
-        </span>
-      </div>
-      <div className="col">
-        <span>PROGRESS</span>
-        <span>
-          Update Progress&nbsp;
-          {progress}
-          %
-        </span>
+
+class Book extends Component {
+  constructor(props) {
+    super(props);
+    const {
+      category, title, author, progress,
+    } = props.book;
+    this.state = {
+      category,
+      title,
+      author,
+      progress,
+      editmode: false,
+    };
+    this.handleEditmode = this.handleEditmode.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleEditmode(event) {
+    const {
+      editmode, category, title, author, progress,
+    } = this.state;
+    document.querySelectorAll('.Book').forEach(book => book.classList.toggle('unfocused'));
+    event.target.parentNode.parentElement.parentElement.classList.remove('unfocused');
+    this.setState({
+      editmode: !editmode,
+      category,
+      title,
+      author,
+      progress,
+    });
+  }
+
+  handleUpdate() {
+    const { updatemethod, deleteindex, book } = this.props;
+    const {
+      category, title, author, progress, editmode,
+    } = this.state;
+    updatemethod({
+      id: book.id, category, title, author, progress,
+    }, deleteindex);
+    document.querySelectorAll('.Book.unfocused').forEach(i => i.classList.remove('unfocused'));
+    this.setState({
+      category, title, author, progress, editmode: !editmode,
+    });
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: value,
+    });
+  }
+
+  render() {
+    const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+    const {
+      category, title, author, progress, editmode,
+    } = this.state;
+    const { deleteindex, handleRemoveBook } = this.props;
+    return (
+      <div className="notshown Book">
+        <div className="col">
+          <span className="fw-700">{category}</span>
+          <span>{title}</span>
+          <span>{author}</span>
+          <span className="sidemenu">
+            <span role="button" tabIndex={0} onClick={() => { handleRemoveBook(deleteindex); }}>Remove</span>
+            <span role="button" tabIndex={0} onClick={this.handleEditmode}>Edit</span>
```

Comment:
`OPTIONAL` I would change these to buttons or apply a `noselect` CSS property so when hovering over these two elements it shows the arrow cursor and not the text selector delimeter.

Reply:
thanks 👍